x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/execute_job/index.ts

```diff
@@ -18,44 +19,41 @@ import {
   getCustomLogo,
 } from '../../../common/execute_job/';
 
-export function executeJobFactory(server) {
+export const executeJobFactory: ExecuteJobFactory<
+  JobDocPayloadPDF,
+  ESQueueWorkerExecuteFn<JobDocPayloadPDF>
+> = function executeJobFactoryFn(server: ServerFacade) {
   const generatePdfObservable = generatePdfObservableFactory(server);
   const logger = LevelLogger.createForServer(server, [PLUGIN_ID, PDF_JOB_TYPE, 'execute']);
 
-  return function executeJob(jobId, jobToExecute, cancellationToken) {
+  return function executeJob(
+    jobId: string,
+    jobToExecute: JobDocPayloadPDF,
+    cancellationToken: any
+  ) {
     const jobLogger = logger.clone([jobId]);
 
-    const process$ = Rx.of({ job: jobToExecute, server }).pipe(
+    const process$ = Rx.of({ job: jobToExecute, server, logger }).pipe(
       mergeMap(decryptJobHeaders),
-      catchError(err => {
-        jobLogger.error(err);
-        return Rx.throwError(
-          i18n.translate(
-            'xpack.reporting.exportTypes.printablePdf.compShim.failedToDecryptReportJobDataErrorMessage',
-            {
-              defaultMessage:
-                'Failed to decrypt report job data. Please ensure that {encryptionKey} is set and re-generate this report. {err}',
-              values: { encryptionKey: 'xpack.reporting.encryptionKey', err: err.toString() },
-            }
-          )
-        );
-      }),
       map(omitBlacklistedHeaders),
       map(getConditionalHeaders),
       mergeMap(getCustomLogo),
       mergeMap(getFullUrls),
-      mergeMap(({ job, conditionalHeaders, logo, urls }) => {
-        return generatePdfObservable(
-          jobLogger,
-          job.title,
-          urls,
-          job.browserTimezone,
-          conditionalHeaders,
-          job.layout,
-          logo
-        );
-      }),
-      map(buffer => ({
+      mergeMap(
+        ({ job, conditionalHeaders, logo, urls }): Rx.Observable<Buffer> => {
+          const { browserTimezone, layout } = job as JobDocPayloadPDF;
```

Comment:
Considering all the mapping above, I fear the `as ` could mask a mistaken type.
Can we avoid the `as` and just specify the type on the signature so that it typeschecks correctly? 🤔

Reply:
Sure, this is simplified when I replace `job` with `jobToExecute`. Sorry, there is a little bit of aliasing variables as they come into the `mergeMap`. For some reason, the job object has to be passed through the operators. This will be cleaned up soon: https://github.com/elastic/kibana/pull/48588

Reply:
40484082b8